source/codegen/templates/proto_helpers.mako

```diff
@@ -53,6 +53,49 @@ enum ${enum_name} {
 % endfor
 </%def>
 
+## Define enums in the proto per attribute type that uses enum values
+<%def name="define_per_type_attribute_enums(attribute_enums)">\
+<%
+  config = data["config"]
+  service_class_prefix = config["service_class_prefix"]
+  enums = data["enums"]
+%>\
+% for type_name in attribute_enums:
+<%
+  type_enums = attribute_enums[type_name]
+  type_enum_name = service_class_prefix + type_name + "Enum"
```

Comment:
I think a better name would be `service_class_prefix + type_name + "AttributeValues"`

Reply:
Updated enum name